### PR TITLE
Add strip_prefix attribute and flag.

### LINKED ISSUE #13: There's no way to differentiate between multiple same-named files across your repo.
If a skylark_doc definition contains two targets with the same filename, the resulting zip has a duplicate file and is challenging to extract.

### PR DESCRIPTION
**Pull request by davidzchen**
Add a strip_prefix attribute for removing a common directory prefix from
generated rule docs.

Fixes #13

### CODE REVIEW

**Comment by davidzchen** [comment]
+cc @garlicnation

File: skydoc/common.py [hunk 1]
```diff
@@ -25,6 +25,12 @@
 OUTPUTS_HEADING = "Outputs:"
 
 
+class InputError(Exception):
+  """Exception raised for errors in the input to Skydoc."""
```
Comment (kchodorow) [inline skydoc/common.py]: Just pass (http://stackoverflow.com/questions/1319615/proper-way-to-declare-custom-exceptions-in-modern-python)

File: skydoc/common_test.py [hunk 2]
```diff
@@ -182,6 +182,21 @@ def test_outputs(self):
...
+
+  def test_strip_prefix(self):
+    bzl_files = ['foo/bar/bar.bzl', 'foo/bar/baz/baz.bzl']
+    self.assertEqual('', common.validate_strip_prefix('', bzl_files))
+    self.assertEqual('foo/', common.validate_strip_prefix('foo/', bzl_files))
+    self.assertEqual('foo/bar/',
+                     common.validate_strip_prefix('foo/bar', bzl_files))
+
+    self._invalid_prefix('bar', ['foo/bar.bzl'])
+    self._invalid_prefix('bar', ['foo/foo.bzl', 'bar/bar.bzl'])
```
Comment (kchodorow) [inline skydoc/common_test.py]: I think `self._invalid_prefix('bar', ['bar.bzl'])` is an interesting test case, add?

Comment (davidzchen) [inline skydoc/common_test.py]: Done.

File: skydoc/common.py (hunk 1 above)
Comment (davidzchen) [inline skydoc/common.py]: Done.

**Comment by davidzchen** [comment]
Jenkins, test this please.

**Comment by davidzchen** [comment]
Looks like Jenkins is having trouble connecting to GitHub: 

```
stderr: fatal: unable to access 'https://github.com/bazelbuild/skydoc/': Failed to connect to github.com port 443: Operation timed out
```

**Comment by damienmg** [comment]
Jenkins retest this please

**Comment by davidzchen** [comment]
PTAL.

**Comment by davidzchen** [comment]
Thanks!

The new GitHub code review features look pretty awesome. :)

**Pull request merged**